lib/protocol/bufferpool_test.go

```diff
@@ -70,18 +70,44 @@ func TestStressBufferPool(t *testing.T) {
 	}
 
 	const routines = 10
-	const runtime = 2 * time.Second
+	ctx, cancel := context.WithTimeout(context.Background(), 2*time.Second)
+	defer cancel()
 
 	bp := newBufferPool()
-	t0 := time.Now()
 
-	var wg sync.WaitGroup
+	checkDone := func() bool {
+		if bp.puts == 0 || bp.skips == 0 || bp.misses == 0 {
```

Comment:
Reading these while the test is running seems like a race condition

Reply:
I thought that was ok because they are atomics. Anyway if, as you noted in the issue, it isn't uncommon for non-short test to use that much memory, this PR can just be closed.